The job **`aarch64-gnu-llvm-20-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/18509066379/job/52745338235) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/52745338235) [(plain)](https://github.com/rust-lang/rust/commit/853c99a0460889326f29a10d5d207d1d8fd05c29/checks/52745338235/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Doc path: /checkout/obj/build/aarch64-unknown-linux-gnu/doc/alloc/index.html
[TIMING:end] doc::Std { build_compiler: Compiler { stage: 2, host: aarch64-unknown-linux-gnu, forced_compiler: false }, target: aarch64-unknown-linux-gnu, format: Html, crates: [] } -- 58.694
##[group]Testing stage2 rustdoc-js-std (aarch64-unknown-linux-gnu)
Testing alias-1.js ... OK
Testing alias-2.js ... TypeError: Cannot read properties of undefined (reading 'pathWithoutLast')
    at handleAlias (/checkout/obj/build/aarch64-unknown-linux-gnu/doc/static.files/search-ca977a4a.js:1:59572)
Bootstrap failed while executing `--stage 2 test --skip compiler --skip src`
Command `/usr/bin/node /checkout/src/tools/rustdoc-js/tester.js --crate-name std --resource-suffix 1.92.0 --doc-folder /checkout/obj/build/aarch64-unknown-linux-gnu/doc --test-folder /checkout/tests/rustdoc-js-std` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/test.rs:1008:27
Executed at: src/bootstrap/src/core/build_steps/test.rs:1035:17

Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:46:17
  local time: Tue Oct 14 21:16:32 UTC 2025
  network time: Tue, 14 Oct 2025 21:16:32 GMT
##[error]Process completed with exit code 1.
##[group]Run echo "disk usage:"
```

</details>

--- a/src/librustdoc/html/static/js/search.js
+++ b/src/librustdoc/html/static/js/search.js
@@ -3933,16 +3933,25 @@ class DocSearch {
                  * @returns {Promise<rustdoc.PlainResultObject?>}
                  */
                 const handleAlias = async(name, alias, dist, index) => {
+                    const item = nonnull(await this.getRow(alias, false));
+                    // space both is an alias for ::,
+                    // and is also allowed to appear in doc alias names
+                    const path_dist = name.includes(" ") || parsedQuery.elems.length === 0 ?
+                        0 : checkRowPath(parsedQuery.elems[0].pathWithoutLast, item);
+                    // path distance exceeds max, omit alias from results
+                    if (path_dist === null) {
+                        return null;
+                    }
                     return {
                         id: alias,
                         dist,
-                        path_dist: 0,
+                        path_dist,
                         index,
                         alias: name,
                         is_alias: true,
                         elems: [], // only used in type-based queries
                         returned: [], // only used in type-based queries
-                        item: nonnull(await this.getRow(alias, false)),
+                        item,
                     };
                 };
                 /**

--- a/tests/rustdoc-js/alias-path-distance-146214.js
+++ b/tests/rustdoc-js/alias-path-distance-146214.js
@@ -3,7 +3,17 @@
 // consider path distance for doc aliases
 // regression test for <https://github.com/rust-lang/rust/issues/146214>
 
-const EXPECTED = {
-    'query': 'Foo::zzz',
-    'others': [{ 'path': 'alias_path_distance::Foo', 'name': 'baz' }],
-};
+const EXPECTED = [
+    {
+        'query': 'Foo::zzz',
+        'others': [
+            { 'path': 'alias_path_distance::Foo', 'name': 'baz' },
+        ],
+    },
+    {
+        'query': '"Foo::zzz"',
+        'others': [
+            { 'path': 'alias_path_distance::Foo', 'name': 'baz' },
+        ],
+    },
+];

